server/src/main/java/org/opensearch/index/query/functionscore/FunctionScoreQueryBuilder.java

```diff
@@ -704,4 +706,12 @@ private static String parseFiltersAndFunctions(
         }
         return currentFieldName;
     }
+
+    @Override
+    public void visit(QueryBuilderVisitor visitor) {
+        visitor.accept(this);
+        if (query != null) {
+            visitor.getChildVisitor(BooleanClause.Occur.MUST).accept(query);
+        }
```

Comment:
Nitpick -- it looks like the `FunctionScoreQueryBuilder` constructors guarantee that `query` can never be `null`, so the check is unnecessary.

Reply:
A similar comment was made [here](https://github.com/opensearch-project/OpenSearch/pull/14739/files#r1681334397), and the author opted to keep the null check to pass tests and keep consistent with the other implementations. Let me know and I can change it if you feel otherwise.

Reply:
I think for unit test case purpose (Keeping method isolated) a null check would be fine